Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docrepository/adapter/model/DocumentMetadata.java

```diff
@@ -0,0 +1,677 @@
+    }
+
+    public String getTypeCode() {
+        return typeCode;
+    }
+
+    public void setTypeCode(String typeCode) {
+        this.typeCode = typeCode;
+    }
+
+    public String getTypeCodeDisplayName() {
+        return typeCodeDisplayName;
+    }
+
+    public void setTypeCodeDisplayName(String typeCodeDisplayName) {
+        this.typeCodeDisplayName = typeCodeDisplayName;
+    }
+
+    public String getTypeCodeScheme() {
+        return typeCodeScheme;
+    }
+
+    public void setTypeCodeScheme(String typeCodeScheme) {
+        this.typeCodeScheme = typeCodeScheme;
+    }
+
+    public boolean isOnDemand() {
+        return onDemand;
+    }
+
+    public void setOnDemand(boolean onDemand) {
+        this.onDemand = onDemand;
+    }
+
+    public String getNewDocumentUniqueId() {
+        return newDocumentUniqueId;
+    }
+
+    public void setNewDocumentUniqueId(String newDocumentUniqueId) {
+        this.newDocumentUniqueId = newDocumentUniqueId;
+    }
+
+    public String getNewRepositoryUniqueId() {
+        return newRepositoryUniqueId;
+    }
+
+    public void setNewRepositoryUniqueId(String newRepositoryUniqueId) {
+        this.newRepositoryUniqueId = newRepositoryUniqueId;
+    }
+
+    public boolean isPersistent() {
+        return persistent;
+    }
+
+    public void setPersistent(boolean persistent) {
+        this.persistent = persistent;
+    }
+
+    public byte[] getRawData() {
+        return document != null ? document.getRawData() : new byte[0];
+    }
+
+    /**
+     * Getter for a code representing the persistent state of the document. This accessor is only to be used with the
+     * persistence layer. Use the isPersistent method to access this property.
+     *
+     * @return Persistence layer code for persistent flag.
+     */
+    public int getPersistentCode() {
+        return persistent ? 1 : 0;
+    }
+
+    /**
+     * Setter for the document's persistence code that is used by the persistence layer. Use the isPersistent method to
+     * look up this property.
+     *
+     * @param persistentCode Persistence layer code for the persistence flag.
+     */
+    public void setPersistentCode(int persistentCode) {
+        persistent = persistentCode == 1;
+
+    }
+
+    public Set<EventCode> getEventCodes() {
+        return eventCodes;
+    }
+
+    public void setEventCodes(Set<EventCode> eventCodes) {
+        this.eventCodes = eventCodes;
+    }
+
+    public Long getPatientRecordId() {
+        return patientRecordId;
+    }
+
+    public void setPatientRecordId(Long patientRecordId) {
+        this.patientRecordId = patientRecordId;
+    }
+
+    public String getPatientIdentifier() {
+        return patientIdentifier;
+    }
+
+    public void setPatientIdentifier(String patientIdentifier) {
+        this.patientIdentifier = patientIdentifier;
+    }
+
+    public String getStatusDisplay() {
+        if (status != null) {
+            String[] aStatus = status.split(":");
```

Comment:
what happen when status doesn't have ":"?  it may throw exception

Reply:
See above comment. I renamed the class, and Git thinks I created the entire file. I left it the way it was.